### ISSUE TITLE
CLI throwing error needs restart

### ISSUE THREAD

**Issue by olivierkrener** [firebase/firebase-tools#2163]
### [REQUIRED] Environment info

**firebase-tools:** 8.1.1

**Platform:** Windows 10

### [REQUIRED] Test case

Have tried to recreate the issue, but have not yet found out, hopefully the logs will help

### [REQUIRED] Steps to reproduce

No way to reproduce it yet as mentioned above

### [REQUIRED] Expected behavior

The CLI should continue working

### [REQUIRED] Actual behavior

When it happends, none of the commands I usually use work anymore:
```
firebase deploy --only hosting
firebase deploy --only firestore:rules
firebase firestore:indexe
```
...

The output seems to be the same with all commands (adding --debug)

> [2020-04-23T19:29:19.561Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
> [2020-04-23T19:29:19.564Z] > authorizing via signed-in user
> [2020-04-23T19:29:19.568Z] [iam] checking project fireauthentest for permissions ["datastore.indexes.list","firebase.projects.get"]
> [2020-04-23T19:29:19.597Z] >>> HTTP REQUEST POST https://cloudresourcemanager.googleapis.com/v1/projects/fireauthentest:testIamPermissions
>  {"permissions":["datastore.indexes.list","firebase.projects.get"]}
> [2020-04-23T19:29:19.658Z] Error: EPERM: operation not permitted, stat

**Comment by mbleigh** [comment]
The issue seems to be that the CLI is trying to read a file to which the current user does not have permission. You'd need to make sure that the current user has full permissions on the working directory as well as on the `$HOME/.config/configstore` directory (sorry I don't know for sure what the Windows equivalent of that is).

**Comment by olivierkrener** [comment]
The current user has full permission on everything, both locally (my window user is an admin for my laptop) and on Firestore (the user I use to login into the CLI is an owner of the project).

As I wrote the problem appears after using the CLI for some time and until I restart it and everything works again. My internet is not the best and it could be that the problem is related to my connection going off sometimes.

Will do my best to understand a way to reproduce it.

**Comment by olivierkrener** [comment]
I have identified that the issue happens only when I have had the emulator running for some time. Then, as soon as I stop the emulator everything is working again as it should. So it seems the emulator is locking some file making them inaccessible to the rest of the CLI
Will get back if I find something to make this reproducible

**Comment by samtstern** [comment]
@olivierkrener thanks for following up.  Your original logs end on this line:
```
[2020-04-23T19:29:19.658Z] Error: EPERM: operation not permitted, stat
```

Do you have the message that comes after that?  That would tell us which file it is having trouble with.

**Comment by olivierkrener** [comment]
I removed it because it showed a local path but forgot to put back the file name, it is the CLI log file

> stat **\firebase-debug.log

**Comment by samtstern** [comment]
@olivierkrener thanks for the info.  So it seems like after ${some long amount of time} we somehow lose access to the debug logs and operations start to randomly fail.

I want to set the right expectations here: this sounds incredibly difficult to reproduce and I don't think it's common enough to warrant a speculatve/defensive fix.  If you see this again it would be interesting if you could check if the `firebase-debug.log` file exists and what the permissions are on it before restarting.

**Comment by olivierkrener** [comment]
>  we somehow lose access to the debug logs and operations start to randomly fail.

Not exactly. The behavior is that after the emulator has been running for some time, any other CLI command will fail due to the error above, meaning the CLI cannot write to the log anymore. As soon as the emulator is stopped, it works again. So it seems that after running for some time the emulator somehow takes an exclusive lock on the log file.

There is a straight forward workaround: stop the emulator, run the commands, and start the emulator after. So this is just an annoying issue but certainly not critical.

Regarding the occurrence of the issue: on my machine it happens all the time provided the emulator runs long enough.

> If you see this again it would be interesting if you could check if the firebase-debug.log file exists and what the permissions are on it before restarting.

Yes the file exists, attached is a snapshot of the security tab on the file properties (I am under Windows 10) when the emulator is running. (I have admin access on this machine but clicking on Advanced wont provide me with more info)
<img width="269" alt="Capture" src="https://user-images.githubusercontent.com/10667327/81164474-b8665c00-8f90-11ea-9e77-4bd98d5d9e3d.PNG">
